src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -45,40 +44,111 @@
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
+import org.joda.time.Duration;
+
 /**
  * Queues up elements until either a duration of time has passed or any threshold in a given set of
  * thresholds is breached, and then delivers the elements in a bundle to the consumer.
  */
 public class ThresholdBundler<E> {
 
-  private final ImmutableList<BundlingThreshold<E>> thresholds;
   private final Duration maxDelay;
+  private ImmutableList<BundlingThreshold<E>> thresholds;
+  private ImmutableList<ExternalThreshold<E>> externalThresholds;
 
   private final Lock lock = new ReentrantLock();
   private final Condition bundleCondition = lock.newCondition();
   private boolean bundleReady = false;
+  private BundleHandle currentBundleHandle;
 
   private Stopwatch bundleStopwatch;
   private final List<E> data = new ArrayList<>();
 
-  public ThresholdBundler(ImmutableList<BundlingThreshold<E>> thresholds) {
-    this(null, thresholds);
+  private ThresholdBundler(Duration maxDelay, ImmutableList<BundlingThreshold<E>> thresholds,
```

Comment:
Nit, list the maxDelay after the thresholds? It seems like the less significant parameter.